Src/WindowsServer/WindowsServer.Shared/FirstChanceExceptionStatisticsTelemetryModule.cs

```diff
@@ -159,7 +153,7 @@ internal bool WasExceptionTracked(Exception exception)
             if (exception.Data != null && !exception.Data.IsReadOnly)
             {
                 // mark exception as tracked
-                exception.Data[ExceptionIsTracked] = null; // The value is unimportant. It's just a sentinel.
+                exception.Data[Thread.CurrentThread.ManagedThreadId] = null; // The value is unimportant. It's just a sentinel.
```

Comment:
Minor: Is something [like this](https://msdn.microsoft.com/en-us/library/dd293602.aspx) possible in .NET code or maybe present in Managed/Unmanaged mix? If exception can be transferred across threads we may detect it couple times.